sim/include/index/entity.h

```diff
@@ -0,0 +1,48 @@
+#ifndef BENTOBOX_ENTITY_H
+#define BENTOBOX_ENTITY_H
+
+#include <unordered_set>
+#include <core/ics/componentStore.h>
+#include <core/ics/util/setIntersection.h>
+
+namespace ics::index {
+
+    class Entity {
```

Comment:
Naming this `Entity` makes think that this represents one Entity instead of a index of all Entities in the System.